### ISSUE TITLE
typings: createInstance should return constructor function

### ISSUE THREAD

**Issue by harbeedev** [ghengeveld/react-async#57]
Hi, I believe the typescript signature of `createInstance` should be:

```ts
export function createInstance<T>(defaultProps?: AsyncProps<T>): new() => Async<T>
```

I have tested this locally, and I'm happy to create a pull request if needed.

**Comment by ghengeveld** [comment]
Yes, that looks good. Better than what we have now at least. Unfortunately it doesn't work with the helper components:

<img width="1055" alt="Schermafbeelding 2019-06-24 om 17 10 12" src="https://user-images.githubusercontent.com/321738/60030290-0341f280-96a3-11e9-937e-5ddbc6f327b5.png">

Please create a PR. Would be great if you can fix the static members as well. I've struggled with typing these in the past and I'm still not comfortable with creating types for that kind of thing.

**Comment by ghengeveld** [comment]
It's not about the "unknown" by the way, if I provide a `promiseFn` that resolves to a `string` it's still the same error (just with `Async<string>`).

**Comment by ghengeveld** [comment]
@harbeedev Will you create a pull request? Let's ignore the subcomponents for now, hopefully somebody will come up with a fix for that.

**Comment by ghengeveld** [comment]
Fixed in [v7.0.4](https://github.com/ghengeveld/react-async/releases/tag/react-async%407.0.4)
